entra_id/go.mod

```diff
@@ -0,0 +1,13 @@
+module entra_id
+
+go 1.18.0
+
+toolchain go1.23.1
```

Comment:
we are not required to specify a `toolchain`. If we specify that this module needs `go 1.18` any available toolchain that is with a higher version will be used. Let's remove this line